src/benchmark.cc

```diff
@@ -212,6 +213,33 @@ void State::FinishKeepRunning() {
   manager_->StartStopBarrier();
 }
 
+ThreadState::ThreadState(State& s) :
+  State(s),
+  parent_(&s)
+{
+  CHECK(!s.started_) << "Don't create a ThreadState object after measurement has started";
+  timer_ = new internal::ThreadTimer(*timer_);
+}
+
+ThreadState::~ThreadState()
+{
+  CHECK(error_occurred() || iterations() >= max_iterations)
+      << "Benchmark returned before ThreadState::KeepRunning() returned false!";
+  {
+    MutexLock l(manager_->GetBenchmarkMutex());
+    internal::MergeResults(*this, timer_, manager_);
+    assert(parent_->total_iterations_ == 0 || parent_->total_iterations_ == total_iterations_);
```

Comment:
why `CHECK` above and `assert` here?

Reply:
As far as I understand, a failed CHECK is something a user (aka the benchmark programmer) did wrong (e.g. wrong API usage). A failed assert is something we (aka the framework programmers) did wrong.